crates/rooch-key/src/keystore/account_keystore.rs

```diff
@@ -43,13 +43,12 @@ pub trait AccountKeystore {
 
     fn generate_and_add_new_key(
         &mut self,
+        account_index: u32,
         password: Option<String>,
     ) -> Result<GeneratedKeyPair, anyhow::Error> {
         // load mnemonic phrase from keystore
         let mnemonic = self.get_mnemonic(password.clone())?;
-        let account_index = mnemonic.mnemonic_data.addresses.len() as u32;
```

Comment:
Why is the `mnemonic_data.addresses.len()` incorrect?

Reply:
> Why is the `mnemonic_data.addresses.len()` incorrect?

See #2022 for details.

Cause there's only one address in `mnemonic_data`.

Reply:
Why only one address in mnemonic_data? It also should be fixed

Reply:
> Why only one address in mnemonic_data? It also should be fixed

https://github.com/rooch-network/rooch/pull/950#issuecomment-1760561544

We should discuss it with @baicaiyihao

Reply:
Use `keystore.addresses()` as the account index is ok. If do this, we can remove `MnemonicData.addresses` field to simplify the MnemonicData definition.
Or when creating an account, need to update the new address to `MnemonicData.addresses`, and can keep `MnemonicData.addresses`.
Both implementations can work.

Reply:
> Use `keystore.addresses()` as the account index is ok. If do this, we can remove `MnemonicData.addresses` field to simplify the MnemonicData definition. Or when creating an account, need to update the new address to `MnemonicData.addresses`, and can keep `MnemonicData.addresses`. Both implementations can work.

It sounds like that we can remove MnemonicData.addresses as mentioned in https://github.com/rooch-network/rooch/pull/950#issuecomment-1760542376.

Reply:
@jolestar